Fixes #34987 - Force 1 call on first load for TableIndexPage

ofedoren opened the pull request:
There are two bugs: 
 - a really small one, so I've included the fix here. Action buttons in new TableIndexPage template didn't allow to pass a function as action instead of href.
 - TableIndexPage did 3 same API calls instead of 1 on first load.
 
 The current behaviour can be tested on /models page. I've tried to use the new template in foreman_webhooks, so this fix was mostly tested on https://github.com/theforeman/foreman_webhooks/pull/51.
 
 The bugs are also present in 3.3.

ofedoren committed:
Fixes #34987 - Force 1 call on first load for TableIndexPage

MariaAga commented on webpack/assets/javascripts/react_app/components/PF4/TableIndexPage/TableIndexPage.js:
```diff
@@ -90,7 +92,11 @@ const TableIndexPage = ({
-    setAPIOptions({ ...apiOptions, params });
+    const defaultAPIOptions = { ...apiOptions, params: defaultParams };
+    const newAPIOptions = { ...apiOptions, params };
+    if (!isEqual(defaultAPIOptions, newAPIOptions)) {
```
Can this juts check `isEqual(defaultParams, params)`?

MariaAga commented:
Thanks for this fix! Since the redmine is about the Modals page - now instead of 3 times the API gets called twice (:tada:), because of `initializeModels`, I think this pr can remove this function and its uses to make no duplicate api calls :)

ofedoren commented:
Thanks, @MariaAga! Updated. I've removed only first initialization for /models page. I guess refactoring models' `index.js` file is out of scope for this PR. But I can do that if needed.

MariaAga reviewed (approved):
Each api call is only being made once, thanks!

Ron-Lavi commented:
CP 3.3: https://github.com/theforeman/foreman/pull/9404
